test/e2e/postinstall/syncsets/syncsets_suite_test.go

```diff
@@ -0,0 +1,517 @@
+				Ω(resultConfigMap.Data).Should(Equal(map[string]string{"foo": "baz-json"}))
+
+			})
+
+			It("Test syncSet secretMappings", func() {
+				ctx := context.TODO()
+				By("Getting the pull secret name of clusterdeployment")
+				cd := &hivev1.ClusterDeployment{}
+				err := hiveClient.Get(ctx, client.ObjectKey{Name: clusterName, Namespace: clusterNamespace}, cd)
+				Ω(err).ShouldNot(HaveOccurred())
+				pullSecret := cd.Spec.PullSecretRef.Name
+
+				SyncSetSecretMappings := &hivev1.SyncSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-syncsecret",
+						Namespace: clusterNamespace,
+					},
+					Spec: hivev1.SyncSetSpec{
+						ClusterDeploymentRefs: []corev1.LocalObjectReference{
+							{
+								Name: clusterName,
+							},
+						},
+						SyncSetCommonSpec: hivev1.SyncSetCommonSpec{
+							ResourceApplyMode: "Sync",
+							Secrets: []hivev1.SecretMapping{
+								{
+									SourceRef: hivev1.SecretReference{
+										Name:      pullSecret,
+										Namespace: clusterNamespace,
+									},
+									TargetRef: hivev1.SecretReference{
+										Name:      "test-pull-secret-copy",
+										Namespace: "default",
+									},
+								},
+							},
+						},
+					},
+				}
+				By("Creating a syncSet SecretMappings")
+				err = hiveClient.Create(ctx, SyncSetSecretMappings)
+				Ω(err).ShouldNot(HaveOccurred())
+
+				err = waitForSyncSetInstanceApplied("test-syncsecret", "syncset", clusterNamespace)
+				Ω(err).ShouldNot(HaveOccurred())
+
+				By("Verifying the secret was copied to target cluster")
+				resultSecret := &corev1.Secret{}
+				err = targetClusterClient.Get(ctx, client.ObjectKey{Name: "test-pull-secret-copy", Namespace: "default"}, resultSecret)
+				Ω(err).ShouldNot(HaveOccurred())
+
+			})
+		})
+		Context("Test selectorSynSet", func() {
+			It("Test selectorSynSet of resources,patches,and secretMappings", func() {
+				ctx := context.TODO()
+				By("Getting the pull secret name of clusterdeployment")
+				cd := &hivev1.ClusterDeployment{}
+				err := hiveClient.Get(ctx, client.ObjectKey{Name: clusterName, Namespace: clusterNamespace}, cd)
+				Ω(err).ShouldNot(HaveOccurred())
+				pullSecret := cd.Spec.PullSecretRef.Name
+
+				By(`Set a label "cluster-group: hivecluster" to clusterdeployment`)
+				cdLabels := cd.ObjectMeta.GetLabels()
+				cdLabels["cluster-group"] = "hivecluster"
+				cd.ObjectMeta.SetLabels(cdLabels)
+				hiveClient.Update(ctx, cd)
+				Ω(cd.ObjectMeta.Labels).Should(Equal(cdLabels))
+
+				resourceConfigMap := &corev1.ConfigMap{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "ConfigMap",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: "default",
+					},
+					Data: map[string]string{
+						"foo": "bar",
+					},
+				}
+
+				SelectorSyncSet := &hivev1.SelectorSyncSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-selectorsyncset",
+						Namespace: "hive",
```

Comment:
SelectorSyncSets are cluster-scoped. The namespace is not needed (it will be ignored anyway)